lib/sse_test_server/sse_server.ex

```diff
@@ -88,12 +88,17 @@ defmodule SSETestServer.SSEServer do
     update_env(%State{state | sse_endpoints: new_endpoints})
   end
 
-  defp update_env(state) do
-    handlers =
-      state.sse_endpoints
+  defp handlers_for_endpoints(sse_endpoints \\ %{}) do
+    sse_handlers =
+      sse_endpoints
       |> Map.values
       |> Enum.map(&SSEEndpoint.to_handler/1)
       |> Enum.sort
+    sse_handlers ++ [{:_, AddHandler, %AddHandler.State{sse_server: self()}}]
```

Comment:
😮 the rare `++`

Reply:
`++` is usually frowned upon because the implementation requires traversing the left list twice to copy elements to the front of the right list. In this case, the lists are short and there's not much we can do to avoid that, so it's okay. `Enum.map` does a similar thing, so it's better to use `Stream.map` for intermediate stages of a pipeline and only build a list again at the end.